src/platform/windows/misc.cpp

```diff
@@ -464,6 +466,37 @@ bp::child run_unprivileged(const std::string &cmd, boost::filesystem::path &work
       (LPSTARTUPINFOW)&startup_info,
       &process_info);
 
+    if(!ret) {
+      auto error = GetLastError();
+
+
+      if(error == 740) {
+        BOOST_LOG(info) << "Could not execute previous command because it required elevation. Running the command again with elevation, for security reasons this will prompt user interaction."sv;
+        startup_info.StartupInfo.wShowWindow = SW_HIDE;
+        startup_info.StartupInfo.dwFlags     = startup_info.StartupInfo.dwFlags | STARTF_USESHOWWINDOW;
+        std::wstring elevated_command        = L"tools\\elevator.exe ";
+        elevated_command += wcmd;
+
+        // For security reasons, Windows enforces that an application can only have one "interactive thread," responsible for processing input from the user and managing the user interface (UI).
+        // Since UAC prompts are interactive, we can't have a UAC prompt while Sunshine is already running because it would block the thread.
+        // To workaround this, we launch a separate process that will prompt the user for elevation and then launch the actual command.
+        // It's pretty much a workaround for restriction on Windows.
+        // Since user has to confirm the UAC prompt, we are retaining a secure method of elevation.
+
+        ret = CreateProcessAsUserW(shell_token,
+          NULL,
+          (LPWSTR)elevated_command.c_str(),
+          NULL,
+          NULL,
+          !!(startup_info.StartupInfo.dwFlags & STARTF_USESTDHANDLES),
+          EXTENDED_STARTUPINFO_PRESENT | CREATE_UNICODE_ENVIRONMENT | CREATE_NEW_CONSOLE | CREATE_BREAKAWAY_FROM_JOB,
+          env_block.data(),
+          start_dir.empty() ? NULL : start_dir.c_str(),
```

Comment:
Use `nullptr` in place of all the `NULL`

Reply:
That's fine, I was just matching the current existing code. Usually, it is better to stay in convention in that regard. It would look sort of strange to swap to nullptr in place of NULL when the code above it is still using NULL.

Should we change all of it, including the original functions code to that?

Reply:
> Should we change all of it, including the original functions code to that?

Eventually yes, but better in a separate PR.